astropy/visualization/wcsaxes/wcsapi.py

```diff
@@ -0,0 +1,252 @@
+        axis_unit = u.Unit(wcs.world_axis_units[idx])
+        coord_wrap = None
+        format_unit = axis_unit
+
+        coord_type = 'scalar'
+
+        if axis_type is not None:
+
+            axis_type_split = axis_type.split('.')
+
+            if "pos.helioprojective.lon" in axis_type:
+                coord_wrap = 180.
+                format_unit = u.arcsec
+            elif "pos.helioprojective.lat" in axis_type:
+                format_unit = u.arcsec
+            elif "pos" in axis_type_split:
+                if "lon" in axis_type_split:
+                    coord_type = "longitude"
+                elif "lat" in axis_type_split:
+                    coord_type = "latitude"
+                elif "ra" in axis_type_split:
+                    coord_type = "longitude"
+                    format_unit = u.hourangle
+                elif "dec" in axis_type_split:
+                    coord_type = "latitude"
+                elif "alt" in axis_type_split:
+                    coord_type = "longitude"
+                elif "az" in axis_type_split:
+                    coord_type = "latitude"
+                elif "long" in axis_type_split:
+                    coord_type = "longitude"
+
+        coord_meta['type'].append(coord_type)
+        coord_meta['wrap'].append(coord_wrap)
+        coord_meta['format_unit'].append(format_unit)
+        coord_meta['unit'].append(axis_unit)
+
+        # For FITS-WCS, for backward-compatibility, we need to make sure that we
+        # provide aliases based on CTYPE for the name.
+        if is_fits_wcs:
+            if isinstance(wcs, WCS):
+                alias = wcs.wcs.ctype[idx][:4].replace('-', '').lower()
+            else:  # SlicedLowLevelWCS
+                alias = wcs._wcs.wcs.ctype[wcs._world_keep[idx]][:4].replace('-', '').lower()
+            name = (axis_type, alias) if axis_type else alias
+        else:
+            name = axis_type or ''
+
+        coord_meta['name'].append(name)
+
+    coord_meta['default_position'] = [''] * wcs.world_n_dim
+
+    m = wcs.axis_correlation_matrix.copy()
+    if invert_xy:
+        m = m[:,::-1]
+
+    if frame_class is RectangularFrame:
+
+        for i, spine_name in enumerate('bltr'):
+            pos = np.nonzero(m[:, i % 2])[0]
+            if len(pos) > 0:
+                coord_meta['default_position'][pos[0]] = spine_name
+                m[pos[0], :] = 0
+
+    elif frame_class is EllipticalFrame:
+
+        if 'longitude' in coord_meta['type']:
+            lon_idx = coord_meta['type'].index('longitude')
+            coord_meta['default_position'][lon_idx] = 'h'
+
+        if 'latitude' in coord_meta['type']:
+            lat_idx = coord_meta['type'].index('latitude')
+            coord_meta['default_position'][lat_idx] = 'c'
+
+    return transform, coord_meta
+
+
+def wcsapi_to_celestial_frame(wcs):
+    for cls, args, kwargs in wcs.world_axis_object_classes.values():
+        if cls is SkyCoord:
```

Comment:
will it always be `SkyCoord`? There is no reason why it couldn't be `BaseCoordinateFrame` right?

Reply:
Yes that's true!

Reply:
Also, shouldn't we use `issubclass` rather than `is` here?

Reply:
@astrofrog @Cadair Is this function the reason you want to implement `world_axis_object_classes` in gwcs? If so what do you think about adding a property `output_frame` to the high-level WCS API? This function implements it in FITS. And it's already available in gwcs. I think this is a totally reasonable addition to the APE as the output frame should always be available through CTYPE. I would even consider `input_frame` but that may require changes to the methods.

Reply:
Changed to issubclass and added support for BaseCoordinateFrame

Reply:
> If so what do you think about adding a property output_frame to the high-level WCS API?

This might be a good idea, although the `HighLevelWeapper` object would probably just use this method of extracting it from the low level object anyway, so we would still want to correctly implement this in gWCS.

*https://github.com/spacetelescope/gwcs/pull/230 is the gWCS PR for this for the record.*